crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -827,8 +854,13 @@ impl<'a, Provider: ResolverProvider> Resolver<'a, Provider> {
                 };
                 let package_id = dist.package_id();
 
-                // If the package does not exist in the registry, we cannot fetch its dependencies
-                if self.unavailable_packages.get(package_name).is_some() {
+                // If the package does not exist in the registry or locally, we cannot fetch its dependencies
+                if self.unavailable_packages.get(package_name).is_some()
+                    && self
+                        .installed_packages
+                        .get_packages(package_name)
+                        .is_empty()
```

Comment:
I believe so but want to confirm: do you have any tests in which a package is already installed, but doesn't exist on the registry? I.e., exercising true for `self.unavailable_packages.get(package_name).is_some()` and false for the latter condition.

Reply:
Yes definitely.